internal/primitive/routed_input.go

```diff
@@ -0,0 +1,281 @@
+		// TODO: #refactor figure out how to extract notifier name... we currently don't pass this into the step when parsing (only notifies/integrations - even those don't have names just types & details)
+		notifierName = "workspace_owners"
+	}
+
+	opts := parseOptionsFromInput(i)
+	payload := NewRoutedInputHttpPayload(
+		r.ExecutionID,
+		r.PipelineExecutionID,
+		r.StepExecutionID,
+		notifierName,
+		map[string]RoutedInputFormData{r.GetShortStepName(): *NewRoutedInputHttpPayloadInput(prompt, inputType, opts)})
+
+	output, err := r.execute(ctx, payload)
+	if err != nil {
+		return nil, err
+	}
+
+	return output, nil
+}
+
+func (r *RoutedInput) execute(ctx context.Context, payload *RoutedInputCreatePayload) (*modconfig.Output, error) {
+	output := &modconfig.Output{}
+
+	if payload == nil {
+		return nil, perr.BadRequestWithMessage("missing payload")
+	}
+
+	// TODO: #refactor #question is this a requirement for all routed inputs? Will they always go to Pipes?
+	token := os.Getenv("FLOWPIPE_PIPES_TOKEN")
+	if token == "" {
+		return nil, perr.InternalWithMessage("missing token")
+	}
+
+	client := &http.Client{}
+
+	id, err := r.initialCreate(ctx, client, token, payload)
+	if err != nil {
+		return nil, err
+	}
+
+	r.Poll(ctx, client, token, id)
+
+	return output, nil
+}
+
+func (r *RoutedInput) initialCreate(ctx context.Context, client *http.Client, token string, payload *RoutedInputCreatePayload) (string, error) {
+	jsonPayload, err := json.Marshal(payload)
+	if err != nil {
+		return "", perr.InternalWithMessage("failed to marshal payload")
+	}
+
+	req, err := http.NewRequest("POST", r.RoutedUrl, bytes.NewBuffer(jsonPayload))
+	if err != nil {
+		return "", perr.InternalWithMessage("failed to create request")
+	}
+
+	req.Header.Set("Authorization", "Bearer "+token)
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", perr.InternalWithMessage("failed to execute request")
+	}
+	defer resp.Body.Close()
+
+	// TODO: #refactor see if we can use some standard shared struct(s)
+	resBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", perr.InternalWithMessage("failed to read response body")
+	}
+
+	var response RoutedInputResponse
+	err = json.Unmarshal(resBody, &response)
+	if err != nil {
+		return "", perr.InternalWithMessage("failed to unmarshal response body")
+	}
+
+	return response.ID, nil
+}
+
+func (r *RoutedInput) Poll(ctx context.Context, client *http.Client, token string, inputID string) {
+	go func() {
+		pollUrl, _ := url.JoinPath(r.RoutedUrl, inputID)
+		req, _ := http.NewRequest("GET", pollUrl, nil)
+		req.Header.Set("Authorization", "Bearer "+token)
+
+		for {
+			var err error
+			time.Sleep(5 * time.Second) // TODO: #refactor better approach - this is at loop initialisation to handle continue from err delay before retry
+
+			resp, err := client.Do(req)
+			if err != nil {
+				// TODO: #error handle errors in polling?
```

Comment:
We can do some sort of back off retry here and terminate the poller after the limit is reached.

Reply:
Yup, I've made some notes for returning to this later, just going to get this to develop for the demo.